alloc.c

```diff
@@ -366,6 +366,12 @@ STATIC void GC_clear_a_few_frames(void)
 /* Heap size at which we need a collection to avoid expanding past      */
 /* limits used by blacklisting.                                         */
 STATIC word GC_collect_at_heapsize = (word)(-1);
+STATIC GC_bool GC_should_trigger_collection = FALSE;
+
+GC_API void GC_trigger_collection()
```

Comment:
I think the naming is slightly odd as this method doesn't actually trigger a collection? Is there a reason you want to set this bool here and then separately call collect?

Reply:
Good points. There isn't really any good reason not to trigger a collection right away. So I'm doing that now, and also made the API name more explicit (and made it only do something in the incremental case).